test/testunusedvar.cpp

```diff
@@ -459,6 +459,20 @@ class TestUnusedVar : public TestFixture {
             "}");
         TODO_ASSERT_EQUALS("error", "", errout.str());
 
+        // constructor with hidden definition
+        functionVariableUsage(
+            "class B {\n"
+            "public:\n"
+            "   B();\n"
+            "};\n"
+            "class A {\n"
+            "   B* b = new B;\n"
+            "};\n"
+            "int main() {\n"
+            "   A a;\n"
+            "}");
+        TODO_ASSERT_EQUALS("error", "", errout.str());
```

Comment:
The assert looks wrong. Looks like a copy and paste error from the one above - and that also looks fishy.

Reply:
Let me check it. Thank you

Reply:
https://github.com/danmar/cppcheck/pull/5764

Reply:
The one above is actually ok as a TODO, since that constructor has no side effects (`class F`).

Reply:
Okay

Reply:
> The one above is actually ok as a TODO, since that constructor has no side effects (`class F`).

"fishy" in the sense that the expected message is just bogus. I did not look at the contents of it.